mobile/src/feeds/home-feeds-screen.tsx

```diff
@@ -1,28 +1,62 @@
 import React from 'react';
-import { FlatList, Text, View} from 'react-native';
+import { FlatList, Text, View, StyleSheet, ListRenderItemInfo } from 'react-native';
 import { TopNavBar } from '../topNavBar/top-nav-bar';
 
-const navBar =  new TopNavBar('Feeds');
-const dataList = [
-{key: 'First Feed'},
-{key: 'Second Feed'},
-{key: 'Third Feed'},
-{key: 'Fourth Feed'},
-{key: 'Fifth Feed'},
-{key: 'Sixth Feed'},
-{key: 'Seventh Feed'},
-{key: 'Ten'}
-];
+import { Employee } from '../reducers/organization/employee.model';
+import { EmployeesStore } from '../reducers/organization/employees.reducer';
+import { Feed } from '../reducers/feeds/feed.model';
+import { connect } from 'react-redux';
+import { AppState } from '../reducers/app.reducer';
 
+import { FeedListItem } from './feed';
 
-export class HomeFeedsScreen extends React.Component {
+import { screenStyles as styles } from './styles';
+
+const navBar = new TopNavBar('Feeds');
+
+interface FeedsScreenProps {
+    feeds: Feed[];
+    employees: EmployeesStore;
+}
+
+const mapStateToProps = (state: AppState): FeedsScreenProps => ({
+    feeds: state.feeds,
+    employees: state.organization.employees
+});
+
+class HomeFeedsScreenImpl extends React.Component<FeedsScreenProps> {
     public static navigationOptions = navBar.configurate();
 
     public render() {
         return (
-            <View>
-                <FlatList data = {dataList} renderItem ={({item}) => <Text style={{padding: 10, fontSize: 18, height: 81, textAlign: 'center'}}>{item.key}</Text>} />
-            </View>
+            <FlatList
+                style={styles.view}
+                keyExtractor={this.keyExtractor}
```

Comment:
why function?

Reply:
In this particular case it should be a function because there should be a unique key for every data item